compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEDmaUtils.cpp

```diff
@@ -39,6 +64,27 @@ bool areAccessPatternsCombinable(const SmallVector<OpFoldResult> &offsetsA,
   if (offsetsA.size() == offsetsB.size() && offsetsA.size() + 1 > maxNbDims)
     return false;
 
+  // Equality of the last N elements of the access patterns of A and B with N =
+  // abs(sizeB - sizeA) results in some simple cases in which the access
+  // patterns are combinable. Note that abs(sizeB - sizeA) should <= 1 and this
+  // is checked for earlier, so just asserted here.
+  if (offsetsA.size() <= offsetsB.size()) {
```

Comment:
Why must `size(A) - 1 <= size(B) < size(A) `, can you not merge whenever B matches a  suffix of A (or A matches a suffix of B)? Assuming compatible offsets that is

Reply:
There is no case with `sizeA > sizeB` and the suffix being completely the same that can be combined.